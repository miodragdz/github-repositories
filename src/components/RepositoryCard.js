import React from 'react'

const RepositoryCard = ({ repository }) => (
  <div className="cards__item">
    <div className="card">
      <div className="card__content">
        <a className="card__name" href={repository.html_url} target="_blank" rel="noopener noreferrer">{repository.full_name}</a>
        <div className="card__language">{repository.language ? repository.language : 'No Language Info'}</div>
        <p className="card__owner">{`Owner: `}
          <a className="card__owner-link" href={repository.owner.html_url} target="_blank" rel="noopener noreferrer">{`${repository.owner.login}`}</a>
        </p>      
        <span className="card__stars">{`Stars: ${repository.stargazers_count}`}</span>
      </div>
    </div>
  </div>
)

export default RepositoryCard
