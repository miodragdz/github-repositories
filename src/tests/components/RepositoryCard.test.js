import React from 'react';
import { shallow } from 'enzyme';
import RepositoryCard from '../../components/RepositoryCard';
import { repositories } from '../fixtures/repositories';

test('should correctly render RepositoryCard with language info', () => {
  const wrapper = shallow(<RepositoryCard key={repositories.items[0].id} repository={repositories.items[0]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render RepositoryCard without language info', () => {
  const wrapper = shallow(<RepositoryCard key={repositories.items[1].id} repository={repositories.items[1]}/>);
  expect(wrapper).toMatchSnapshot();
});
