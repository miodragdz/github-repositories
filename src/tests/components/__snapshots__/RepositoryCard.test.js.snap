// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render RepositoryCard with language info 1`] = `ShallowWrapper {}`;

exports[`should correctly render RepositoryCard without language info 1`] = `ShallowWrapper {}`;
